department-of-veterans-affairs/vets-website
# Pension- Update single-page multiresponse styles

## Summary

- Our implementation of the multi response pattern does not adhere to VADS guidance. This is causing display and accessibility issues. The new design outlined in [#78336](https://github.com/department-of-veterans-affairs/va.gov-team/issues/78336) solves for the issues identified by the Platform governance team (issues detailed in [#73984](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73984), [#73983](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73983), [#73981](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73981)).
- The new pattern and its interactive states are applied to all sections of the form that use a single-page multi-response pattern, including other service names, current/previous jobs, VA/federal medical centers, dependent children, income sources, and care/medical expenses.
- This is behind the [pension_multiresponse_styles flipper](https://staging-api.va.gov/flipper/features/pension_multiresponse_styles). See https://github.com/department-of-veterans-affairs/vets-api/pull/16423. 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/79329
- [Flipper creation](https://github.com/department-of-veterans-affairs/vets-api/pull/16423)

## Testing done

- Disable the feature flag, refresh the site, and run through the entire form, confirming that there are no visible changes from previous behavior and styling. 
- Disable the feature flag, refresh the site, and run through the entire form, filling out each of the multi-response pages listed above. For each page, add multiple items, edit items, and remove items. Leave at least two items when moving on to the next section. On the review page, edit, add, and remove items in each of the sections. Confirm that the behavior and styles on the pages match [the designs](https://www.figma.com/file/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?type=design&node-id=2759-18967&mode=design&t=XXVJ0nFvmp32MpNX-0). 

## Screenshots

|         | Before | After |
| ------- | ------ | ----- |
| Multi-line edit  |     <img width="434" alt="Screenshot 2024-04-22 at 11 51 02 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/ea2a5299-c8cf-405b-8a38-b54078bf92a4"> | <img width="546" alt="Screenshot 2024-04-22 at 11 49 41 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/c23a8121-acc5-4442-84d1-d2ccb2102b7d">  |
| Errors | <img width="350" alt="Screenshot 2024-04-22 at 9 41 05 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/26b58bf1-3371-4172-9c6a-331df83b766e"> | <img width="359" alt="Screenshot 2024-04-22 at 9 40 18 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/6d99e8ac-9be0-4b4b-bc1e-316fb259d2bd"> |
| First Item | <img width="560" alt="Screenshot 2024-04-22 at 9 31 42 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/a8ef781a-b6d8-45fc-ba94-c5b238e08177">  | <img width="540" alt="Screenshot 2024-04-22 at 9 39 43 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/21046714/cc77bacc-bb0c-4c53-969c-598a8b17bc68"> |

## What areas of the site does it impact?

Pension Benefits

## Acceptance criteria

- initial list item card aligned left with header and other form elements
- 16px card padding
- error state left border alignment matches designs
- edit button on collapsed card with details aligned lower right
- add save button to new list item card

### Quality Assurance & Testing


### Error Handling


### Authentication
